### PR TITLE
shouldVisitChild: Check parent and child node previously deferred separately

### PR DESCRIPTION
**Pull request by wbinnssmith**
Follows up and fixes an issue in #7035 that would result in "Asset was skipped or not found." by unmarking asset group parents when the asset group was deferred, not the dependency.

Test Plan:
* [x] Test in large app
* [x] Integration test

### CODE REVIEW

File: packages/core/core/src/AssetGraph.js [hunk 1]
```diff
@@ -304,14 +304,15 @@ export default class AssetGraph extends ContentGraph<AssetGraphNode> {
 
     let {sideEffects, canDefer = true} = childNode.value;
     let dependency = node.value;
-    let previouslyDeferred = node.hasDeferred;
+    let nodePreviouslyDeferred = node.hasDeferred;
+    let childNodePreviouslyDeferred = childNode.deferred;
```
Comment (mischnic) [inline packages/core/core/src/AssetGraph.js]: Should we rename them to `dependencyPreviouslyDeferred` and `assetGroupPrev....`?

(This is ensured by the if statement above)

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.74s | -40.00ms |
| Cached | 288.00ms | -17.00ms 🚀 |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/index.96016b08.js | 1.59kb | +0.00b | 771.00ms | -60.00ms 🚀 |
| dist/legacy/index.8aaa89c9.js | 1.20kb | +0.00b | 771.00ms | -59.00ms 🚀 |
| dist/modern/index.6be20f01.js | 1.13kb | +0.00b | 770.00ms | -60.00ms 🚀 |
| dist/legacy/index.html | 826.00b | +0.00b | 890.00ms | -62.00ms 🚀 |
| dist/modern/index.html | 749.00b | +0.00b | 889.00ms | -63.00ms 🚀 |
| dist/legacy/index.c1bc86aa.css | 94.00b | +0.00b | 1000.00ms | -76.00ms 🚀 |
| dist/modern/index.57a95cbe.css | 94.00b | +0.00b | 999.00ms | -76.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 9.22s | -464.00ms 🚀 |
| Cached | 519.00ms | +70.00ms ⚠️ |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 487.00kb | +0.00b | 4.87s | -287.00ms 🚀 |
| dist/PermalinkedComment.62775cc6.js | 4.20kb | +0.00b | 4.87s | -287.00ms 🚀 |
| dist/UserProfile.1b7befb5.js | 1.57kb | +0.00b | 4.87s | -287.00ms 🚀 |
| dist/NotFound.e6c89571.js | 429.00b | +0.00b | 4.87s | -286.00ms 🚀 |
| dist/logo.c5bb83f1.png | 246.00b | +0.00b | 4.82s | -292.00ms 🚀 |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.03m | +1.03m ⚠️ |
| Cached | 1.66s | +1.66s ⚠️ |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.36s | -61.00ms |
| Cached | 373.00ms | -10.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/312284395861967424)**

File: packages/core/core/src/AssetGraph.js (hunk 1 above)
Comment (wbinnssmith) [reply packages/core/core/src/AssetGraph.js]: I had the same thought. I kept these consistent with what existed, but it's probably best just to move everything to this.

**Review by mischnic — APPROVED** [review]

**Comment by mischnic** [comment]
Asset graph of the test case. Notice that dependency at the bottom left is marked "hasDeferred" (so deferred) but the symbol "bar" is actually used. This caused the "Asset was skipped or not found" error

AssetGraph_Main

**Review by devongovett — APPROVED** [review]

**Pull request merged**